pkg/cloudprovider/providers/azure/azure_vmss.go

```diff
@@ -654,27 +680,66 @@ func (ss *scaleSet) EnsureHostsInPool(serviceName string, nodes []*v1.Node, back
 	return nil
 }
 
-// EnsureBackendPoolDeleted ensures the loadBalancer backendAddressPools deleted from the specified vmSet.
-func (ss *scaleSet) EnsureBackendPoolDeleted(poolID, vmSetName string) error {
-	virtualMachineScaleSet, exists, err := ss.getScaleSetWithRetry(vmSetName)
+// EnsureHostsInPool ensures the given Node's primary IP configurations are
+// participating in the specified LoadBalancer Backend Pool.
+func (ss *scaleSet) EnsureHostsInPool(serviceName string, nodes []*v1.Node, backendPoolID string, vmSetName string, isInternal bool) error {
+	scalesets, standardNodes, err := ss.getNodesScaleSets(nodes)
+	if err != nil {
+		glog.Errorf("getNodesScaleSets() for service %q failed: %v", serviceName, err)
+		return err
+	}
+
+	for ssName, instanceIDs := range scalesets {
+		// Only add nodes belonging to specified vmSet to basic LB backends.
+		if !ss.useStandardLoadBalancer() && ssName != vmSetName {
```

Comment:
Need to check `vmSetName != "" ` like others? We'd better add a note what does it mean for `vmSetName==""`

Reply:
Also use !strings.EqualFold to align with Line438?

Reply:
ACK